wire-ios-request-strategy/Sources/Synchronization/Decoding/StoreUpdateEvent.swift

```diff
@@ -95,6 +101,34 @@ public final class StoredUpdateEvent: NSManagedObject {
         return storedEvent
     }
 
+    static private func storedEventExists(for updateEvent: ZMUpdateEvent, in context: NSManagedObjectContext) -> Bool {
+        guard let eventId = updateEvent.uuid?.transportString() else {
+            assertionFailure("trying to check storedEvent without id")
+            return false
+        }
+
+        let storedEvents = StoredUpdateEvent.events(eventId: eventId, context: context)
+        guard !storedEvents.isEmpty else {
+            return false
+        }
+
+        let upateEventHash = EventHasher(eventId: eventId, payload: updateEvent.payload as NSDictionary).hashValue
+
+        for storedEvent in storedEvents {
+            let storedEventHash = EventHasher(storedEvent: storedEvent)?.hashValue
+            return storedEventHash == upateEventHash
```

Comment:
This would return in the first iteration. Perhaps you mean something like:

```swift
return storedEvents.contains {
    storedEventHash == EventHasher(storedEvent: storedEvent)?.hashValue
}
```

Reply:
changed it